microsoft/testsuites/kdump/kdumpcrash.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from lisa import (
+    Node,
+    SkippedException,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+    schema,
+)
+
+from pathlib import PurePosixPath
+from lisa.tools import Dmesg, Uname, Kdump
+from assertpy import assert_that
+from lisa.operating_system import Redhat, Suse, Debian
+
+
+@TestSuiteMetadata(
+    area="kdump",
+    category="functional",
+    description="""
+    This test suite is used to check if vmcore is generated after triggering
+    kdump through injecting NMI or echoing c into sysrq-trigger.
+    """,
+)
+class KdumpCrash(TestSuite):
+    crash_kernel = "512M"
+    vmbus_string = "Vmbus version:"
+    dump_path = "/var/crash"
+    trigger_kdump_cmd = "sync; echo c > /proc/sysrq-trigger"
+    config_path = ""
+
+    def _get_boot_config_path(self, node: Node) -> str:
+        uname_tool = node.tools[Uname]
+        kernel_ver = uname_tool.get_linux_information().kernel_version
+        config_path = f"/boot/config-{kernel_ver}"
+        assert_that(node.shell.exists(PurePosixPath(config_path))).described_as(
+            f"/boot/config-{kernel_ver} not exist."
+            " Please check if the kernel version is right."
+        ).is_true()
+        return config_path
+
+    def _kdump_test(self, node: Node) -> None:
+        dmesg = node.tools[Dmesg]
+        if self.vmbus_string not in dmesg.get_output():
+            raise SkippedException(
+                "Negotiated VMBus version is not 3.0. "
+                "Kernel might be old or patches not included. "
+                "Full support for kdump is not present."
+            )
```

Comment:
write a similar method _get_vmbus_version
in https://github.com/microsoft/lisa/blob/main/lisa/sut_orchestrator/azure/platform_.py#L572?

Reply:
Seems that they are private functions. We couldn't use it in test cases.

Reply:
The logic to get vmbus version from dmesg, it should be implemented in Dmesg tool. Avoid to overuse grep in code. It hides details, hard to troubleshooting failures.

Reply:
Added get_vmbus_version function in dmesg tool.